ci/src/test-python.py

```diff
@@ -9,69 +10,187 @@
         return json.load(f)
 
-def get_latest_plugin(manifest):
+def get_latest_plugin(manifest: dict) -> dict:
+    """Get latest plugin from manifest."""
+    untested_plugins = []
     for _plugin in manifest[::-1]:
         if _plugin["Language"] == "python" and "Tested" not in _plugin.keys():
-            if "github.com" not in _plugin["UrlSourceCode"]:
-                print("Non-Github based website!")
-                sys.exit(0)
-            break
-    else:
-        print("No Untested plugin found!")
+            untested_plugins.append(_plugin)
+    if len(untested_plugins) == 0:
+        print_section("Test failed!", "No Untested plugin found!\nTest could not find a plugin without \"Tested\" key.")
+        sys.exit(1)
+    if len(untested_plugins) > 1:
+        print_section("Test failed!", "More than one untested plugin found!")
         sys.exit(1)
-    return _plugin
+    return untested_plugins[0]
 
-def run_plugin(plugin_path, execute_path):
+def run_plugin(plugin_name: str, plugin_path: str, execute_path: str) -> None:
+    """Run plugin and check output."""
     os.chdir(plugin_path)
-    p = Popen(["python", "-S", Path(Path(plugin_path, execute_path)), '{\"method\": \"query\", \"parameters\": [\"\"]}'], text=True, stdout=PIPE, stderr=PIPE)
+    default_settings = init_settings(plugin_name, plugin_path)
+    args = json.dumps(
+        {"method": "query", "parameters": [""], "Settings": default_settings}
+    )
+    full_args = ["python", "-S", Path(plugin_path, execute_path), args]
+    # Older Flox used environmental variable to locate Images directory
+    os.environ["PYTHONPATH"] = str(USER_PATH.joinpath("PythonEmbeddable"))
+    print_section("Input", full_args)
+    p = Popen(full_args, text=True, stdout=PIPE, stderr=PIPE)
     stdout, stderr = p.communicate()
     exit_code = p.wait()
     if stdout != "":
-        print(f'{"#" * 9} Output {"#" * 9}\n{stdout}')
-    if exit_code == 0:
-        print("Test passed!")
+        print_section("Output", stdout)
+        valid_json, json_msg = test_valid_json(stdout)
+    if exit_code == 0 and valid_json:
+        print_section("Test passed!", "", repeat_times=20)
     else:
-        print(f'Test failed!\nPlugin returned a non-zero exit code!\n{"#" * 9} Trace {"#" * 9}')
+        print(f'Test failed!\nPlugin returned a non-zero exit code!')
         if stderr != "":
-            print(stderr)
-        sys.exit(exit_code)
+            print_section('Trace', stderr)
+        if json_msg:
+            print(json_msg)
+        sys.exit(max(exit_code, 1))
+
+
+def setup_flow_environment() -> None:
+    """Setup Flow Launcher local & roaming directory."""
+    _mkdir(USER_PATH)
+    _mkdir(APP_PATH)
+    for _dir in USER_DIRS:
+        _mkdir(Path(USER_PATH, _dir))
+    for _dir in APP_DIRS:
+        _mkdir(Path(APP_PATH, _dir))
+    os.makedirs(Path(USER_PATH, "Settings", "Plugins"), exist_ok=True)
+    os.makedirs(Path(APP_PATH, "app-1.9.0"), exist_ok=True)
```

Comment:
this will still hold when the version is no longer 1.9.0?

Reply:
Yeah I don't think this will affect anything.

Reply:
cool, let's use a generic version.
```suggestion
    os.makedirs(Path(APP_PATH, "app-1.0.0"), exist_ok=True)
```